src/components/text-editor/menu/menu-filter.ts

```diff
@@ -0,0 +1,107 @@
+import { Dropdown, MenuElement, MenuItem } from 'prosemirror-menu';
+import { EditorButton } from './types';
+
+/**
+ * The available items for the editor toolbar
+ * The values must match the titles of the menuItem
+ *
+ * @see MenuItem.spec
+ */
+const allItems: Record<EditorButton, string> = {
+    strong: 'Toggle strong style',
+    emphasis: 'Toggle emphasis',
+    code: 'Toggle code font',
+    link: 'Add or remove link',
+    'horizontal rule': 'hr',
+    paragraph: 'Change to paragraph',
+    'code block': 'Change to code block',
+    'heading 1': 'Change to heading 1',
+    'heading 2': 'Change to heading 2',
+    'heading 3': 'Change to heading 3',
+    'heading 4': 'Change to heading 4',
+    'heading 5': 'Change to heading 5',
+    'heading 6': 'Change to heading 6',
+    undo: 'Undo last change',
+    redo: 'Redo last undone change',
+    'bullet list': 'Wrap in bullet list',
+    'ordered list': 'Wrap in ordered list',
+    'block quote': 'Wrap in block quote',
+    'join above': 'Join with above block',
+    'lift out': 'Lift out of enclosing block',
+};
+
+const defaultItems: EditorButton[] = [
+    'strong',
+    'emphasis',
+    'code',
+    'link',
+    'horizontal rule',
+    'paragraph',
+    'code block',
+    'heading 1',
+    'heading 2',
+    'heading 3',
+    'heading 4',
+    'heading 5',
+    'heading 6',
+    'block quote',
+    'bullet list',
+    'ordered list',
+    'join above',
+    'lift out',
```

Comment:
```suggestion
    'strong',
    'emphasis',
    'link',
    'heading 1',
    'heading 2',
    'heading 3',
    'heading 4',
    'heading 5',
    'heading 6',
    'block quote',
    'bullet list',
    'ordered list',
```

The above suffice for "our" default set (assuming that all the headings will be bundled in our own <kbd>H ˯</kbd> dropdown menu.

Advanced users can always use key-bindings. Consumers with specific needs to visualize more buttons in the UI can always add more. 👌